### ISSUE TITLE
Unable to set set hubName using host.json

### ISSUE THREAD

**Issue by mathiasi** [Azure/azure-functions-durable-extension#2602]
### Description
I'm getting the `Microsoft.Azure.WebJobs.Extensions.DurableTask: Task Hub name must be specified in host.json when using slots.` error in my staging slot however I have tried to set value using host.json as per the instructions with environtment variables. This is my host.json as it has been deployed in the slot:
```json
D:\home\site\wwwroot>cat host.json
{
  "version": "2.0",
  "logging": {
    "logLevel": {
      "Function.HealthCheck": "Warning"
    },
    "applicationInsights": {
      "LogLevel": {
        "Default": "Information"
      }
    },
    "extensions": {
      "durableTask": {
        "hubName": "%TaskHubName%"
      }
    }
  }
}
```
And this is my configuration:
![image](https://github.com/Azure/azure-functions-durable-extension/assets/2315801/f555d38b-056c-4840-ba3b-56f46750b133)

I have been searching and seen some cases of people overriden IConfiguration but we are not messing with that in our Startup.cs. My personal theory is that we are using IWebJobsStartup and not FunctionsStartup but it isn't easy for us to switch and test right now.

### Expected behavior
Error to disappear and the defined name to be used. 

### Actual behavior
Error persists

### App Details

- **Durable Functions extension version (e.g. v1.8.3)**: 2.10.0 
- **Azure Functions runtime version (1.0 or 2.0)**: 2.0
- **Programming language used**: C#

Function is running as In-Process.

### If deployed to Azure
> We have access to a lot of telemetry that can help with investigations. Please provide as much of the following information as you can to help us investigate!

- **Timeframe issue observed**:
- **Function App name**:
- **Function name(s)**:
- **Azure region**:
- **Orchestration instance ID(s)**:
- **Azure storage account name**:

> If you don't want to share your Function App or storage account name GitHub, please at least share the orchestration instance ID. Otherwise it's extremely difficult to look up information.

**Comment by lilyjma** [comment]
Hi @mathiasi - could you check[ this comment ](https://github.com/Azure/azure-functions-durable-extension/issues/1057#issuecomment-692171425)and see if #1 applies?

**Comment by mathiasi** [comment]
Hi @lilyjma - unfortunately it does not. I went with something completely different to make sure.

**Comment by bachuv** [comment]
Hi @mathiasi, are you still experiencing this issue? I wasn't able to reproduce it on my end. Are you by chance overriding any configuration setup in your Startup.cs?

**Comment by mathiasi** [comment]
Hi @bachuv 
In the meanwhile we changed from in-process to isolated and I jut went back to revisit this. I was still seeing some issues but it seems to be related to the name I was using even though it was within the specification (alphanumeric etc.) when I changed to simple names such as "mainhub" and "slothub" then it started working again.
I believe there might still be something that can be improved but this is at least no longer an issue for us.
